core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -844,6 +909,61 @@ class MultiCloudWorkspaceService(override val ctx: RawlsRequestContext,
     }
   }
 
+  private def updateWorkspaceRecordWithResources(workspaceId: UUID, cloudPlatform: CloudPlatform): Future[Unit] = {
+    if (cloudPlatform != CloudPlatform.GCP) {
+      logger.info("No updates for non-GCP")
+      return Future.successful()
+    }
+
+    for {
+      maybeWorkspace <- workspaceRepository.getWorkspace(workspaceId)
+      workspace <- Future {
+        maybeWorkspace.getOrElse(throw new RuntimeException("Workspace not found"))
+      }
+      wsmWorkspace <- Future {
+        workspaceManagerDAO.getWorkspace(workspaceId, ctx)
+      }
+      _ <- Future {
+        workspaceRepository.updateBucketName(workspace.workspaceIdAsUUID,
```

Comment:
It looks like these 2 `workspaceRepository` methods already return `Future` so don't need to be wrapped in `Future`